fbg-server/src/rooms/rooms.service.ts

```diff
@@ -98,23 +99,51 @@ export class RoomsService {
       if (room.match) {
         return room;
       }
-      const userMembership = room.userMemberships.find(
-        (membership) => membership.user.id === userIdOfCaller,
-      );
-      if (!userMembership || !userMembership.isCreator) {
+      await this.checkIsOwner(userIdOfCaller, room);
+      await this.removeMembership(queryRunner, userIdToBeRemoved, room);
+      if (room.isPublic) {
+        await this.lobbyService.notifyLobbyUpdate();
+      }
+      return room;
+    });
+  }
+
+  /** Updates room metadata (capacity and game). */
+  async updateRoom(
+    userIdOfCaller: number,
+    updateRoomInput: UpdateRoomInput
+  ): Promise<void> {
+    await inTransaction(this.connection, async (queryRunner) => {
+      const room = await this.getRoomEntity(updateRoomInput.roomId);
+      if (room.match) {
         throw new HttpException(
-          'You must be the creator of the room',
+          'Game already started for this Room',
           HttpStatus.BAD_REQUEST,
         );
       }
-      await this.removeMembership(queryRunner, userIdToBeRemoved, room);
+      await this.checkIsOwner(userIdOfCaller, room);
+      room.capacity = updateRoomInput.capacity;
+      room.gameCode = updateRoomInput.gameCode;
+      await queryRunner.manager.save(room);
+      await this.notifyRoomUpdate(room);
       if (room.isPublic) {
         await this.lobbyService.notifyLobbyUpdate();
       }
-      return room;
     });
   }
 
+  private async checkIsOwner(userId, room: RoomEntity) {
+    const userMembership = room.userMemberships.find(
+        (membership) => membership.user.id === userId,
+    );
+    if (!userMembership || !userMembership.isCreator) {
```

Comment:
Using TypeScript's [optional chaining operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining), this could be just
```typescript 
if (!userMembership?.isCreator) {
```